src/authMiddleware.ts

```diff
@@ -0,0 +1,58 @@
+import { ZodiosRouterContextRequestHandler } from "@zodios/express";
+import { match, P } from "ts-pattern";
+import { ExpressContext } from "./app.js";
+import { readClaimsFromJwtToken } from "./auth/jwt.js";
+import { CatalogProcessError, ErrorCode } from "./model/domain/errors.js";
+import { ApiError, mapAuthorizationErrorToApiError } from "./model/types.js";
+
+export const authMiddleware: ZodiosRouterContextRequestHandler<
+  ExpressContext
+> = (req, res, next) => {
+  try {
+    const authorization = req.headers.authorization;
+    return match(authorization)
+      .with(P.string, (auth: string) => {
+        const bearer = auth.replace("Bearer ", "");
+        if (!bearer) {
```

Comment:
Why is this check useful? It's not actually checking if `auth` starts with `Bearer`, so I'm not sure the error message is correct

More general point: do we want to restrict strings (and non-boolean values) in general in boolean expressions? Checks like `!b` are often dangerous in JS and adding some restrictions with eslint could be helpful especially considering we're migrating from a language where this is not a concern. So maybe should we use https://typescript-eslint.io/rules/strict-boolean-expressions? (We've used it in other projects.) @gabro what do you think?